From 8188377e0874cd039c2cf906d45eb6606ce74ca1 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 2 Feb 2024 15:37:21 +0100
Subject: [PATCH 1010/1695] drivers: stm32_remoteproc: remove temporary
 st,s_memory-region property

Remove the use of type based on the st,s_memory-region that will be
replaced by a management of the memory access based on the firewall.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d88557e51120267fa90106783a98776a35e9528a drivers: stm32_remoteproc: remove temporary st,s_memory-region property]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ic91b261cf7f53db991fa9f7dd85cf1f9bc988a38
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391541
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 45 +++++-----------------
 1 file changed, 9 insertions(+), 36 deletions(-)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index 1d3749f5f..46a9f8e57 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -30,13 +30,11 @@
  * @da:		device address corresponding to the physical base address
  *		from remote processor space perspective
  * @size:	size of the region
- * @type:	specify if the memory is secure or not secure
  */
 struct stm32_rproc_mem {
 	paddr_t addr;
 	paddr_t da;
 	size_t size;
-	enum teecore_memtypes type;
 };
 
 /**
@@ -111,10 +109,6 @@ static TEE_Result stm32mp2_rproc_start(struct stm32_rproc_instance *rproc)
 			continue;
 
 #if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
-		if ((!rproc->tzen && mems[i].type == MEM_AREA_RAM_SEC) ||
-		    (rproc->tzen && mems[i].type == MEM_AREA_RAM_NSEC))
-			return	TEE_ERROR_GENERIC;
-
 		if (rproc->tzen) {
 			stm32mp_syscfg_write(A35SSC_M33_INITSVTOR_CR,
 					     rproc->boot_addr, INITVTOR_MASK);
@@ -231,10 +225,9 @@ TEE_Result stm32_rproc_da_to_pa(uint32_t rproc_id, paddr_t da, size_t size,
 	return TEE_ERROR_ACCESS_DENIED;
 }
 
-static TEE_Result stm32_rproc_map_mem(paddr_t pa, size_t size, void **va,
-				      enum teecore_memtypes type)
+static TEE_Result stm32_rproc_map_mem(paddr_t pa, size_t size, void **va)
 {
-	*va = core_mmu_add_mapping(type, pa, size);
+	*va = core_mmu_add_mapping(MEM_AREA_RAM_NSEC, pa, size);
 	if (!*va) {
 		EMSG("Can't map region %#"PRIxPA" size %zu", pa, size);
 		return TEE_ERROR_GENERIC;
@@ -259,28 +252,19 @@ TEE_Result stm32_rproc_map(uint32_t rproc_id, paddr_t pa, size_t size,
 		if (!core_is_buffer_inside(pa, size, mems[i].addr,
 					   mems[i].size))
 			continue;
-		/*
-		 * Only secure part is tested, the non-secure is not tested
-		 * as it is not possible to distinguish the FW memory from
-		 * the shared memory. The shared memory could be used by
-		 * the secure firmware.
-		 */
-		if (mems[i].type == MEM_AREA_RAM_SEC && !rproc->tzen)
-			return	TEE_ERROR_CORRUPT_OBJECT;
 
-		return stm32_rproc_map_mem(pa, size, va, mems[i].type);
+		return stm32_rproc_map_mem(pa, size, va);
 	}
 
 	return TEE_ERROR_ACCESS_DENIED;
 }
 
-static TEE_Result stm32_rproc_unmap_mem(void *va, size_t size,
-					enum teecore_memtypes type)
+static TEE_Result stm32_rproc_unmap_mem(void *va, size_t size)
 {
 	/* Flush the cache before unmapping the memory */
 	dcache_clean_range(va, size);
 
-	if (core_mmu_remove_mapping(type, va, size)) {
+	if (core_mmu_remove_mapping(MEM_AREA_RAM_NSEC, va, size)) {
 		EMSG("Can't unmap region %p size %zu", va, size);
 		return TEE_ERROR_GENERIC;
 	}
@@ -305,7 +289,7 @@ TEE_Result stm32_rproc_unmap(uint32_t rproc_id, void *va, size_t size)
 					   mems[i].size))
 			continue;
 
-		return stm32_rproc_unmap_mem(va, size, mems[i].type);
+		return stm32_rproc_unmap_mem(va, size);
 	}
 
 	return TEE_ERROR_ACCESS_DENIED;
@@ -372,11 +356,11 @@ TEE_Result stm32_rproc_clean(uint32_t rproc_id)
 	for (i = 0; i < rproc->n_regions; i++) {
 		pa = mems[i].addr;
 		size = mems[i].size;
-		res = stm32_rproc_map_mem(pa, size, &va, mems[i].type);
+		res = stm32_rproc_map_mem(pa, size, &va);
 		if (res)
 			return res;
 		memset(va, 0, size);
-		res = stm32_rproc_unmap_mem(va, size, mems[i].type);
+		res = stm32_rproc_unmap_mem(va, size);
 		if (res)
 			return res;
 	}
@@ -449,7 +433,6 @@ static TEE_Result stm32_rproc_parse_mems(struct stm32_rproc_instance *rproc,
 
 	for (i = 0; i < n_regions; i++) {
 		int pnode = 0;
-		uint32_t sec_mem = 0;
 
 		pnode = fdt_node_offset_by_phandle(fdt, fdt32_to_cpu(list[i]));
 		if (pnode < 0) {
@@ -469,22 +452,12 @@ static TEE_Result stm32_rproc_parse_mems(struct stm32_rproc_instance *rproc,
 		if (res)
 			goto err;
 
-		/* TODO remove temporary property and use firewall */
-		res = fdt_read_uint32_index(fdt, node, "st,s-memory-region",
-					    i, &sec_mem);
-		if (res)
-			sec_mem = 0;
-
 		if (!regions[i].addr || !regions[i].size) {
 			res = TEE_ERROR_BAD_PARAMETERS;
 			goto err;
 		}
 
-		regions[i].type = sec_mem ?
-				  MEM_AREA_RAM_SEC : MEM_AREA_RAM_NSEC;
-
-		DMSG("register %s region %#"PRIxPA" size %#zx",
-		     sec_mem ? "sec" : " nsec",
+		DMSG("register region %#"PRIxPA" size %#zx",
 		     regions[i].addr, regions[i].size);
 	}
 
-- 
2.47.2

